libobs/obs-source.c

```diff
@@ -1098,15 +1103,36 @@ void obs_source_set_texcoords_centered(obs_source_t *source, bool centered)
 	source->texcoords_centered = centered;
 }
 
+static void check_for_duplicate_audio(const obs_source_t *source, bool active)
```

Comment:
I feel like this function should be renamed, a `check_something` function is usually expected to return a value (the result of the check) and be pure (aka it should not change state).

This function however checks _and_ changes state and does not return the result of the check.

So it should either return `match` with the caller then making the decision whether to change the ref count.

Also it seems like the ref count is increased when a source with a duplicate device is activated, but if the same source is deactivated, the ref count is not decreased (because `active` is `false`) - is that by design?

Reply:
Changed it to update_duplicate_audio_refs

> Also it seems like the ref count is increased when a source with a duplicate device is activated, but if the same source is deactivated, the ref count is not decreased (because active is false) - is that by design?

Since it compares (match && active) the refs will increment if both are true and will decrement if either is false. In testing everything is working as expected.